feature-libs/product-configurator/rulebased/cpq/occ/cpq-configurator-occ.service.ts

```diff
@@ -113,4 +120,224 @@ export class CpqConfiguratorOccService {
+  ): Observable<Configurator.Configuration> {
+    return this.callReadConfigurationForCartEntry(parameters).pipe(
+      this.converterService.pipeable(CPQ_CONFIGURATOR_NORMALIZER)
+    );
+  }
+
+  /**
+   * Retrieves a configuration assigned to an order entry.
+   *
+   * @param {CommonConfigurator.ReadConfigurationFromOrderEntryParameters} parameters - Order entry parameters
+   * @returns {Observable<Configurator.Configuration>} - Retrieved configuration
+   */
+  readConfigurationForOrderEntry(
+    parameters: CommonConfigurator.ReadConfigurationFromOrderEntryParameters
+  ): Observable<Configurator.Configuration> {
+    return this.callReadConfigurationForOrderEntry(parameters).pipe(
+      this.converterService.pipeable(CPQ_CONFIGURATOR_NORMALIZER)
+    );
+  }
+
+  protected callCreateConfiguration(
+    productSystemId: string
+  ): Observable<Cpq.Configuration> {
+    const url = this.occEndpointsService.buildUrl('createCpqConfiguration', {
+      urlParams: {
+        productCode: productSystemId,
+      },
+    });
+    return this.http.get<Cpq.Configuration>(url);
+  }
+
+  protected callReadConfiguration(
+    configId: string,
+    tabId?: string
+  ): Observable<Cpq.Configuration> {
+    const url = this.occEndpointsService.buildUrl('readCpqConfiguration', {
+      urlParams: {
+        configurationId: configId,
+      },
+      queryParams: tabId ? { tabId: tabId } : undefined,
+    });
+    return this.http.get<Cpq.Configuration>(url);
+  }
+
+  protected callReadConfigurationOverview(
+    configId: string
+  ): Observable<Cpq.Configuration> {
+    const url = this.occEndpointsService.buildUrl(
+      'readCpqConfigurationOverview',
+      {
+        urlParams: {
+          configurationId: configId,
+        },
+      }
+    );
+    return this.http.get<Cpq.Configuration>(url);
+  }
+
+  protected callUpdateAttribute(
+    updateAttribute: Cpq.UpdateAttribute
+  ): Observable<Cpq.Configuration> {
+    const url = this.occEndpointsService.buildUrl('updateCpqAttribute', {
+      urlParams: {
+        configurationId: updateAttribute.configurationId,
+        attributeCode: updateAttribute.standardAttributeCode,
+      },
+      queryParams: updateAttribute.tabId
+        ? { tabId: updateAttribute.tabId }
+        : undefined,
+    });
+    return this.http.patch<Cpq.Configuration>(
+      url,
+      updateAttribute.changeAttributeValue
+    );
+  }
+
+  protected callUpdateValue(
+    updateValue: Cpq.UpdateValue
+  ): Observable<Cpq.Configuration> {
+    const url = this.occEndpointsService.buildUrl(
+      'updateCpqAttributeValueQuantity',
+      {
+        urlParams: {
+          configurationId: updateValue.configurationId,
+          attributeCode: updateValue.standardAttributeCode,
+          attributeValueId: updateValue.attributeValueId,
+        },
+        queryParams: updateValue.tabId
+          ? { tabId: updateValue.tabId }
+          : undefined,
```

Comment:
branch not covered - and how shouldthis happen? UpdateAttribute.tabId is not optional, so it shall not be undefined.

Reply:
done